backend/migrations/007_add_multi_source_support.py

```diff
@@ -0,0 +1,56 @@
+"""Add multi-source translation support tables."""
+import peewee as pw
+
+# Use proxy pattern like other migrations to avoid model conflicts
+db_proxy = pw.DatabaseProxy()
+
+class _Base(pw.Model):
+    class Meta:
+        database = db_proxy
+
+class SourceTranslationLinks(_Base):
+    id = pw.AutoField(primary_key=True)
+    origin_source_id = pw.IntegerField()
+    translated_source_id = pw.IntegerField()
+
+    class Meta:
+        database = db_proxy
+        table_name = 'source_translation_links'
+        indexes = (
+            (('origin_source_id',), False),
+            (('translated_source_id',), False),
+        )
+
+class SegmentTranslationLinks(_Base):
+    id = pw.AutoField(primary_key=True)
+    origin_segment_id = pw.IntegerField()
+    origin_segment_timestamp = pw.DateTimeField()
+    translated_segment_id = pw.IntegerField()
+    translated_segment_timestamp = pw.DateTimeField()
+
+    class Meta:
+        database = db_proxy
+        table_name = 'segment_translation_links'
+        indexes = (
+            (('origin_segment_id', 'origin_segment_timestamp'), False),
+            (('translated_segment_id', 'translated_segment_timestamp'), False),
+        )
+
+def migrate(migrator, database, fake=False, **kwargs):
+    db_proxy.initialize(database)
+    
+    # Drop tables if they exist with wrong structure (from previous incorrect migration)
+    # This is safe because these are link tables with no critical data
+    if database.table_exists('source_translation_links'):
```

Comment:
Did you add this just for your debugging iterations?
We don't need this in prod, right?

Remove this.